packages/react/src/dialog/Dialog.tsx

```diff
@@ -0,0 +1,50 @@
+import * as RadixDialog from "@radix-ui/react-dialog";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { CloseIcon } from "./CloseIcon";
+import * as styles from "./Dialog.css";
+
+type DialogProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  ComponentPropsWithRef<typeof RadixDialog.Root>,
+  {
+    children: ReactNode;
+    hideCloseIcon?: boolean;
+  } & styles.DialogVariants
+>;
+export const Dialog = forwardRef<HTMLDivElement, DialogProps>(
+  ({ children, hideCloseIcon = false, size = "md", ...props }, ref) => {
+    return (
+      <Box asChild ref={ref} {...props}>
+        <RadixDialog.Root>
+          <Box asChild>
+            <RadixDialog.Portal>
+              <Flex asChild {...styles.overlay()}>
+                <RadixDialog.Overlay>
+                  <Box asChild {...styles.content({ size })}>
+                    <RadixDialog.Content aria-describedby={undefined}>
+                      {children}
+                      {!hideCloseIcon && (
+                        <Box asChild {...styles.close()}>
+                          <RadixDialog.Close aria-label="Close" asChild>
+                            {CloseIcon}
```

Comment:
let's use a button here - so users can use keyboard to use it
```suggestion
                            <Button appearance="secondary" icon={CloseIcon} />
```